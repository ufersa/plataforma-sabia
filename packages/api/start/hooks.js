const { hooks } = require('@adonisjs/ignitor');

const AlgoliaFakeProvider = require('../test/utils/AlgoliaFakeService');
const BlogFakeService = require('../test/utils/BlogFakeService');

hooks.before.providersRegistered(() => {
	if (process.env.NODE_ENV === 'testing') {
		/*
		|--------------------------------------------------------------------------
		| Add fakes
		|--------------------------------------------------------------------------
		|
		| Fake provider services before instantiation
		|
		*/
		const { ioc } = use('@adonisjs/fold');
		ioc.singletonFake('App/Services/AlgoliaSearch', () => AlgoliaFakeProvider);
		ioc.singletonFake('App/Services/Blog', () => BlogFakeService);
	}
});

hooks.after.providersBooted(() => {
	const Validator = use('Validator');

	// eslint-disable-next-line global-require
	const CustomValidators = require('../app/Utils/CustomValidators');

	Object.values(CustomValidators).forEach((validator) => {
		Validator.extend([validator.name], validator);
	});

	const Config = use('Adonis/Src/Config');
	const View = use('View');

	View.global('APP_URL', Config.get('app.appURL'));
	View.global('WEB_URL', Config.get('app.webURL'));
	View.global('STORAGE_URL', Config.get('upload.storageURL'));
	View.global('PLATFORM_MAIL', Config.get('mail.platform.mail'));
});
