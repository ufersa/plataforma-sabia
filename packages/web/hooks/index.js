export { default as useTheme } from './useTheme';
export { default as useModal } from './useModal';
export { default as useAuth } from './useAuth';
export { default as useTechnology } from './useTechnology';
export { default as useVisibleComponent } from './useVisibleComponent';
export { default as useShoppingCart } from './useShoppingCart';
export { default as useMediaQuery } from './useMediaQuery';
