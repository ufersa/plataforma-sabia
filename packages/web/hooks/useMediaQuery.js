import { useState, useEffect, useCallback } from 'react';

export default function useMediaQuery(width) {
	const [targetReached, setTargetReached] = useState(false);

	const updateTarget = useCallback((e) => {
		if (e.matches) {
			setTargetReached(true);
		} else {
			setTargetReached(false);
		}
	}, []);

	useEffect(() => {
		const media = window.matchMedia(`(max-width: ${width}px)`);
		media.addEventListener('change', (e) => updateTarget(e));

		// Check on mount (callback is not called until a change occurs)
		if (media.matches) {
			setTargetReached(true);
		}

		return () => media.removeEventListener('change', (e) => updateTarget(e));
	}, [updateTarget, width]);

	return targetReached;
}
