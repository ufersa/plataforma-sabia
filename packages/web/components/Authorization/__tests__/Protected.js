/* eslint-disable jsx-a11y/aria-role */
import React from 'react';
import { render, screen } from 'test-utils';
import * as Router from 'next/router';
import { Protected } from '..';
import * as useAuth from '../../../hooks/useAuth';

jest.mock('../../SignInForm', () => ({
	__esModule: true,
	default: () => <div data-testid="sign-in-form" />,
}));

describe('Protected component', () => {
	it('should redirect to login if user is logged out', () => {
		const mockPush = jest.fn();
		jest.spyOn(Router, 'useRouter').mockReturnValue({
			push: mockPush,
			asPath: 'redirecting',
		});
		const childrenText = 'children';

		const { container } = render(
			<Protected>
				<h1>{childrenText}</h1>
			</Protected>,
		);

		expect(container).toMatchSnapshot();
		expect(screen.queryByText(childrenText)).not.toBeInTheDocument();
		expect(mockPush).toHaveBeenCalledTimes(1);
		expect(mockPush).toHaveBeenCalledWith('/entrar?redirect=redirecting');
	});

	it('should render children if user is logged in and the component does not require a role', () => {
		jest.spyOn(useAuth, 'default').mockReturnValue({
			user: {
				email: 'test@test.com',
			},
			isAuthenticated: true,
		});

		const childrenText = 'children';

		const { container } = render(
			<Protected>
				<h1>{childrenText}</h1>
			</Protected>,
		);

		expect(container).toMatchSnapshot();
		expect(container.querySelector('h1').textContent).toEqual(childrenText);
	});

	it('should render children if user is logged in and has the role', () => {
		jest.spyOn(useAuth, 'default').mockReturnValue({
			user: {
				email: 'test@test.com',
				role: {
					role: 'admin',
				},
			},
			isAuthenticated: true,
		});

		const childrenText = 'children';

		const { container } = render(
			<Protected userRole="admin">
				<h1>{childrenText}</h1>
			</Protected>,
		);

		expect(container).toMatchSnapshot();
		expect(container.querySelector('h1').textContent).toEqual(childrenText);
	});

	it('should not render children if user is logged in and does not have the role', () => {
		jest.spyOn(useAuth, 'default').mockReturnValue({
			user: {
				email: 'test@test.com',
				role: {
					role: 'moderator',
				},
			},
			isAuthenticated: true,
		});

		const childrenText = 'children';

		const { container, getByTestId } = render(
			<Protected userRole="admin">
				<h1>{childrenText}</h1>
			</Protected>,
		);

		expect(container).toMatchSnapshot();
		expect(getByTestId('notAuthorized')).toBeInTheDocument();
		expect(container.querySelector('h1').textContent).not.toEqual(childrenText);
	});

	it('should redirect to another page when user is not logged in and component have a redirect prop', () => {
		jest.spyOn(Router, 'useRouter').mockReturnValue({
			push: jest.fn(),
		});

		const childrenText = 'children';
		const redirectTo = '/';

		const { container } = render(
			<Protected userRole="admin" redirectTo={redirectTo}>
				<h1>{childrenText}</h1>
			</Protected>,
		);

		expect(container).toMatchSnapshot();
		expect(container.querySelector('h1').textContent).not.toEqual(childrenText);
		expect(Router.useRouter().push).toHaveBeenCalledWith(redirectTo);
		expect(Router.useRouter().push).toHaveBeenCalledTimes(1);
	});

	it('should show an inline login when inline props was passed', () => {
		jest.spyOn(useAuth, 'default').mockReturnValue({
			user: {
				email: null,
			},
		});

		const childrenText = 'children';

		const { container } = render(
			<Protected inline>
				<h1>{childrenText}</h1>
			</Protected>,
		);

		expect(container).toMatchSnapshot();
		expect(screen.getByTestId(/sign-in-form/i)).toBeInTheDocument();
	});

	it('should return only the inline message when onlyUnauthorizedMessage is provided', () => {
		const childrenText = 'children';

		jest.spyOn(useAuth, 'default').mockReturnValue({
			user: {
				email: null,
			},
		});

		const { container, queryByTestId } = render(
			<Protected onlyUnauthorizedMessage>
				<h1>{childrenText}</h1>
			</Protected>,
		);

		expect(container).toMatchSnapshot();
		expect(container.querySelector('h1').textContent).not.toEqual(childrenText);
		expect(queryByTestId('modal')).not.toBeTruthy();
		expect(queryByTestId('notAuthorized')).toBeTruthy();
		expect(container.querySelector('form input[type=email]')).not.toBeTruthy();
		expect(container.querySelector('form input[type=password]')).not.toBeTruthy();
		expect(container.querySelector('form button[type=submit]')).not.toBeTruthy();
	});
});
