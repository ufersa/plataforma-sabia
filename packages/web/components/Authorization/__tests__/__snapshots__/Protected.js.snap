// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Protected component should not render children if user is logged in and does not have the role 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c0 h1 {
  color: hsla(0,0%,7%);
  font-size: 3.6rem;
  margin: auto;
}

<div>
  <div
    class="c0"
    data-testid="notAuthorized"
  >
    <h1>
      Você não possui autorização para acessar este conteúdo
    </h1>
  </div>
</div>
`;

exports[`Protected component should redirect to another page when user is not logged in and component have a redirect prop 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c0 h1 {
  color: hsla(0,0%,7%);
  font-size: 3.6rem;
  margin: auto;
}

<div>
  <div
    class="c0"
    data-testid="notAuthorized"
  >
    <h1>
      Você não possui autorização para acessar este conteúdo
    </h1>
  </div>
</div>
`;

exports[`Protected component should redirect to login if user is logged out 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c0 h1 {
  color: hsla(0,0%,7%);
  font-size: 3.6rem;
  margin: auto;
}

<div>
  <div
    class="c0"
    data-testid="notAuthorized"
  >
    <h1>
      Você não possui autorização para acessar este conteúdo
    </h1>
  </div>
</div>
`;

exports[`Protected component should render children if user is logged in and has the role 1`] = `
<div>
  <h1>
    children
  </h1>
</div>
`;

exports[`Protected component should render children if user is logged in and the component does not require a role 1`] = `
<div>
  <h1>
    children
  </h1>
</div>
`;

exports[`Protected component should return only the inline message when onlyUnauthorizedMessage is provided 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c0 h1 {
  color: hsla(0,0%,7%);
  font-size: 3.6rem;
  margin: auto;
}

<div>
  <div
    class="c0"
    data-testid="notAuthorized"
  >
    <h1>
      Você não possui autorização para acessar este conteúdo
    </h1>
  </div>
</div>
`;

exports[`Protected component should return the login modal if user is not logged in 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c0 h1 {
  color: hsla(0,0%,7%);
  font-size: 3.6rem;
  margin: auto;
}

<div>
  <div
    class="c0"
    data-testid="notAuthorized"
  >
    <h1>
      Você não possui autorização para acessar este conteúdo
    </h1>
  </div>
</div>
`;

exports[`Protected component should show an inline login when inline props was passed 1`] = `
.c11 {
  position: relative;
  border: 2px solid transparent;
  background: none;
  font-size: 1.4rem;
  font-weight: bold;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.2rem 0.8rem;
  line-height: 2.4rem;
  text-align: center;
  color: hsl(166,100%,34%);
  text-transform: none;
  font-weight: 500;
  border-radius: 0.5rem;
  padding: 0.8rem;
  gap: 0.8rem;
  color: hsla(0,0%,100%);
  background-color: hsl(166,100%,34%);
  text-transform: uppercase;
  font-weight: bold;
}

.c11:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c11:hover:not(:disabled) {
  color: hsla(0,0%,100%);
  background-color: hsl(166,100%,34%);
  border-color: hsl(166,100%,34%);
}

.c11:focus {
  box-shadow: 0px 0px 4px 2px hsla(29,94%,62%);
}

.c11:hover:not(:disabled) {
  opacity: 0.74;
}

.c5 {
  color: hsla(204,7%,15%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c8 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  opacity: 1;
  background: hsla(0,0%,100%);
  border: 1px solid hsl(0,0%,93%);
  border-radius: 0.5rem;
  color: hsla(0,0%,7%);
}

.c8::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c8::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c8:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c8::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c8::-webkit-input-placeholder {
  color: hsl(0,0%,74%);
  font-style: normal;
}

.c8::-moz-placeholder {
  color: hsl(0,0%,74%);
  font-style: normal;
}

.c8:-ms-input-placeholder {
  color: hsl(0,0%,74%);
  font-style: normal;
}

.c8::placeholder {
  color: hsl(0,0%,74%);
  font-style: normal;
}

.c4 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  color: hsla(0,79%,63%);
  margin-left: 0.5rem;
  display: inline-block;
}

.c10 {
  color: hsla(204,70%,53%);
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  font-weight: 500;
}

.c10:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10:focus {
  box-shadow: 0px 0px 4px 2px hsla(29,94%,62%);
}

.c13 {
  color: hsla(204,70%,53%);
  -webkit-align-self: left;
  -ms-flex-item-align: left;
  align-self: left;
  font-weight: 700;
}

.c13:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c13:focus {
  box-shadow: 0px 0px 4px 2px hsla(29,94%,62%);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  max-width: 100%;
  width: 33rem;
}

.c2 {
  line-height: 2.8rem;
  margin-bottom: 3.2rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2.4rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.6rem;
}

.c9 > a {
  margin-bottom: 0.8rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 2.4rem;
  color: hsla(0,0%,31%);
  margin-top: 4.8rem;
}

.c12 > a {
  font-size: 1.6rem;
  margin-left: 0.4rem;
}

@media screen and (min-width:768px) {
  .c11 {
    width: 100%;
  }
}

@media screen and (max-width:1023.9px) {
  .c0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    padding-top: 3.2rem;
  }
}

@media screen and (max-width:1023.9px) {
  .c12 {
    margin-top: 3.2rem;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c12 > a {
    margin-left: 0;
  }
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <h3
        class="c2"
      >
        Faça login com sua conta
      </h3>
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <label
            class="c5"
            for="email"
          >
            E-mail
             
            <span
              aria-label="Indicador visual de campo obrigatório"
              class="c6"
            >
              *
            </span>
          </label>
          <div
            class="c7"
          >
            <input
              aria-label="E-mail"
              aria-required="true"
              class="c8"
              id="email"
              name="email"
              placeholder="Digite o seu e-mail"
              type="text"
            />
          </div>
        </div>
        <div
          class="c4"
        >
          <label
            class="c5"
            for="password"
          >
            Senha
             
            <span
              aria-label="Indicador visual de campo obrigatório"
              class="c6"
            >
              *
            </span>
          </label>
          <div
            class="c7"
          >
            <input
              aria-label="Senha"
              aria-required="true"
              class="c8"
              id="password"
              name="password"
              placeholder="Digite a sua senha"
              type="password"
            />
          </div>
        </div>
      </div>
      <div
        class="c9"
      >
        <a
          class="c10"
          color="blue"
          target="_self"
        >
          Esqueceu sua senha?
        </a>
        <button
          class="c11"
          name=""
          type="submit"
        >
          Entrar
        </button>
      </div>
    </div>
    <div
      class="c12"
    >
      <span>
        Não possui conta? 
      </span>
      <a
        class="c13"
        color="blue"
        target="_self"
      >
        Cadastre-se agora mesmo
      </a>
    </div>
  </form>
</div>
`;
